synapse/neuron.py

```diff
@@ -490,11 +491,10 @@ def onlink(ok, link):
 
             isok, sess = retn.wait(timeout=timeout)
             if not isok:
-                # XXX Raise / log here?
-                return None
+                raise s_common.RetnTimeout(msg='retnwait timed out or failed')
```

Comment:
if isok is false, there will be a excfo tufo in ``sess`` so raising RetnTimeout is incorrect (there could be something else there)